### DISCUSSION TITLE
Updating build section for Windows using Visual Studio

### DISCUSSION THREAD

**Discussion by serianox** [Yubico/libfido2#682] [Ideas]
Hi,

I'm not sure if this is intended, or if it just works - the readme says that one should read the PowerShell build scripts for CI if they intend to build on Windows, and those scripts heavily relies on Cygwin which for some reason doesn't work for me.

However, I was able to build directly using Visual Studio 20xx which directly supports CMake and every other requirements for the project. The process is quite simple, _but_ requires Visual Studio.

Should I update the readme to provide the steps to others?

Best regards,

**Comment by LDVG** [comment]
Hi,

The `windows/build.ps1` script should not require Cygwin (but does require a set of programs to be installed to automatically fetch and build libfido2's dependencies). The [last part of the script](https://github.com/Yubico/libfido2/blob/86b2503abda657ad0044df428f1d14d68a9d44a4/windows/build.ps1#L207-L242) defines  the libfido2 build itself. Was there anything in particular that was confusing?

**Comment by serianox** [comment]
Hmm, maybe I got confused in the Windows build scripts somehow. :(

CMake is required for the build, but is not automatically installed in `windows/build.ps1`. I thought that it was the role of `windows/cygwin.ps1` to solve this (my issue is probably specific to my setup, Cygwin installed through the provided script or manually doesn't work...).

Anyway, the CMake from Visual Studio works great. May I ask which compiler are you using for the Windows release?

**Reply by LDVG** [reply]
> CMake is required for the build, but is not automatically installed in windows/build.ps1. I thought that it was the role of windows/cygwin.ps1 to solve this (my issue is probably specific to my setup, Cygwin installed through the provided script or manually doesn't work...).

Ah, no, these scripts are used entirely separately (both build libfido2). :-)

> Anyway, the CMake from Visual Studio works great. May I ask which compiler are you using for the Windows release?

We're using MSVC for our Windows releases.
